### DISCUSSION TITLE
require errors during package transpiliation on nextjs app (other platform .node not found)

### DISCUSSION THREAD

**Discussion by is-it-ayush** [napi-rs/napi-rs#1858] [Q&A]
Hi! I'm trying to transpile `Package A` (a rust project) that exposes a few functions in my nextjs application (say `Package B`). The problem is, The build stage of nextjs app fails complaining `│ Module not found: Can't resolve '@scope/package_a-linux-arm-gnueabihf` for every other platform (so almost 10 similar errors). My `Package A` is built using `napi build --platform --strip --no-const-enum ./bindings` which generates a single `.node` file. The error makes sense but how should I workaround the fact that the other `require("some_different_platform.node")` is not available during build & just include `.node` files that are available in my nextjs bundle.

One of the workarounds that I'm aware of is to simply not transpile the package at all. In this case, I would have to manually make sure the package is present after compilation when I run `next dev`. Although, This is not what I'm hoping for.

**Comment by sagoez** [comment]
Where you able to find a workaround for this?

**Reply by is-it-ayush** [reply]
Yes @samgj18! The error was caused because napi generated an `index.js` file iirc. I simply wrote that file manually & removed other platform requirement code which napi auto-generated. This allowed the project to compile successfully. It isn't a clean solution since it adds a manual step but works. 
![image](https://github.com/napi-rs/napi-rs/assets/36449128/86367a8e-a02d-4d42-86ec-1a8344454d94)
